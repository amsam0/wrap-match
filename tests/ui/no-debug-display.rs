fn main() {}

struct Error;

#[wrap_match::wrap_match(error_message = "{error:?}")]
fn no_debug() -> Result<(), Error> {
    Err(Error)?;
    Ok(())
}

#[wrap_match::wrap_match(error_message = "{error}", error_message_without_info = "")]
fn no_display() -> Result<(), Error> {
    Err(Error)?;
    Ok(())
}
