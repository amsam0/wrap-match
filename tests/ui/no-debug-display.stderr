error[E0277]: `Error` doesn't implement `std::fmt::Debug`
 --> tests/ui/no-debug-display.rs:5:1
  |
5 | #[wrap_match::wrap_match(error_message = "{error:?}")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Error` cannot be formatted using `{:?}`
  |
  = help: the trait `std::fmt::Debug` is not implemented for `Error`
  = note: add `#[derive(Debug)]` to `Error` or manually `impl std::fmt::Debug for Error`
  = note: this error originates in the macro `format_args` which comes from the expansion of the attribute macro `wrap_match::wrap_match` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Error` with `#[derive(Debug)]`
  |
3 | #[derive(Debug)]
  |

error[E0277]: `Error` doesn't implement `std::fmt::Display`
  --> tests/ui/no-debug-display.rs:11:1
   |
11 | #[wrap_match::wrap_match(error_message = "{error}", error_message_without_info = "")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Error` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `Error`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `format_args` which comes from the expansion of the attribute macro `wrap_match::wrap_match` (in Nightly builds, run with -Z macro-backtrace for more info)
